alpha/apps/kaltura/lib/db/DbManager.php

```diff
@@ -233,35 +244,39 @@ protected static function getStickySessionKey()
 	/**
 	 * @return KalturaPDO
 	 */
-	public static function getSphinxConnection($read = true)
+	public static function getSphinxConnection($read = true, $indexName = null)
 	{
-		if(!self::$sphinxConnection)
+		KalturaLog::debug("Using index with name [$indexName]");
+		if(!isset(self::$sphinxConnection[$indexName]))
 		{
-			$sphinxDS = isset(self::$config['sphinx_datasources']['datasources']) ? self::$config['sphinx_datasources']['datasources'] : array(self::DB_CONFIG_SPHINX);
+			if($indexName && isset(self::$config['sphinx_datasources_'.$indexName]) && isset(self::$config['sphinx_datasources_'.$indexName]['datasources']))
+				$sphinxDS = self::$config['sphinx_datasources_'.$indexName]['datasources'];
+			else
+				$sphinxDS = isset(self::$config['sphinx_datasources']['datasources']) ? self::$config['sphinx_datasources']['datasources'] : array(self::DB_CONFIG_SPHINX);
```

Comment:
if isset(self::$config['sphinx_datasources_'.$indexName]['datasources']) is true, but it has empty value, you will not reach the else.
Better add isset(self::$config['sphinx_datasources_'.$indexName]['datasources']) to the condition above , and if it fails , set array array(self::DB_CONFIG_SPHINX)